### DISCUSSION TITLE
Lazy properties are not supported

### DISCUSSION THREAD

**Discussion by McNight** [pointfreeco/swift-perception#27] [Q&A]
### Description

Hello!

It seems that **lazy properties** are currently not supported.

Let's say that I have the following code:

```swift
@Perceptible
class Feature {
  lazy var magic = 3.14
}
```

I will not be able to compile the project due to the following error:
> 'lazy' cannot be used on a computed property
> FIX: Remove 'lazy '

which (correct me if I'm wrong) but does not seem to be an accurate diagnostic.
The fix will indeed make my code compile but it will not respect the behaviour I wanted to have in the first place.

What do you think should be done for this kind of situation ?

I tried to add a very basic workaround by rejecting lazy properties similar to what you did for immutable properties:
```swift
// in Sources/PerceptionMacros/Extensions.swift
extension VariableDeclSyntax {
  // ...
  var isLazy: Bool {
    for modifier in modifiers {
      for token in modifier.tokens(viewMode: .all) {
        if token.tokenKind == .keyword(.lazy) {
          return true
        }
      }
    }
    return false
  }
}

// in Sources/PerceptionMacros/PerceptibleMacro.swift
extension VariableDeclSyntax {
  // ...
  var isValidForPerception: Bool {
    !isComputed && isInstance && !isImmutable && !isLazy && identifier != nil
  }
}
```

but is it really what should be done?

I'm reporting the issue but based on what you decide, I would gladly help and make the appropriate changes if necessary 😄 

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

### Expected behavior

Either fully support the source code input or reject it but with a more appropriate diagnostic 

### Actual behavior

Not compiling and inaccurate diagnostic

### Perception version information

1.1.1

### Xcode version information

15.2

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: arm64-apple-macosx14.0
```

**Comment by stephencelis** [comment]
@McNight This is an issue that affects Swift's `@Observable`, as well:

```swift
@Observable class Foo {
  lazy var bar = 1  🛑
}
```

So we think if you'd like to fix it it'd be best to do so upstream with a bug filed to [apple/swift](https://github.com/apple/swift).

Since this isn't a bug with our library I'm going to convert it to a discussion.

**Comment by McNight** [comment]
Oh I see, indeed I should have tested with the official Observable implementation as well...
I'll make some research to see if it has been already brought to the table somewhere (on forums) and if something is already planned for it.
